common/autoware_trajectory/include/autoware/trajectory/interpolator/detail/interpolator_common_interface.hpp

```diff
@@ -120,8 +120,9 @@ class InterpolatorCommonInterface
       return static_cast<int32_t>(bases_.size()) - 2;
     }
     auto comp = [](const double & a, const double & b) { return a <= b; };
-    return std::distance(bases_.begin(), std::lower_bound(bases_.begin(), bases_.end(), s, comp)) -
-           1;
+    const int32_t idx =
+      std::distance(bases_.begin(), std::lower_bound(bases_.begin(), bases_.end(), s, comp)) - 1;
+    return std::clamp(idx, 0, bases_.size() - 1);
```

Comment:
I'm sorry, my previous proposal was wrong as [this CI fail](https://github.com/autowarefoundation/autoware_core/actions/runs/16265235628/job/45980723584?pr=568) shows.

I tested the following fix worked on my local environment.

```suggestion
    return std::clamp(idx, static_cast<int32_t>(0), static_cast<int32_t>(bases_.size() - 1));
```